processor/transformprocessor/internal/common/processor.go

```diff
@@ -0,0 +1,415 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/multierr"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottldatapoints"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottllogs"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlresource"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlscope"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottltraces"
+)
+
+type Context interface {
+	// IsContext dummy method for type safety
+	IsContext()
```

Comment:
Make it internal then?

Reply:
I am confused about the need of this func.

Reply:
@bogdandrutu the goal (and maybe this is the wrong goal?) is to have a single `ParseContextStatements` function that knows how to handle any type of context.  To be usable by Trace processors, Metric processors, and Log processors, it needs to be able to return a type that all 3 could use.  For this design I originally had `Context` look like 

```go
type Context interface {
	ProcessTraces(td ptrace.Traces)
	ProcessMetrics(td pmetric.Metrics)
	ProcessLogs(td plog.Logs)
}
```
This worked great for the implementing `ResourceContext` struct and `ScopeContext` struct which can get used in all three signal processors, but for `TraceContext` struct, it meant useless empty functions to implement `ProcessMetrics` and  `ProcessLogs`.  You can see this style in my [first commit](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/15382/commits/07457a88ca89e3f92870403d9d45385343a542f2#diff-9fc1947976347cbd9dfdbfecdbdd447ddbdcb1e86bc380ae0b9bce47956f6e0e).

The current commit's `Context` is 1 step above `interface{}`, giving users confidence that it can be casted to one of `TracesContext`, `MetricsContext`, or `LogsContext` safely.

 I had originally started out locally with a `ParseContextStatements` per signal type, but tried to condense things since there was a lot of duplication.  Unfortunately the different Contexts don't have something natural to tie them all together like the consumer package's `Capabilities()` function on `baseConsumer`.  Open to suggestions.

Reply:
@bogdandrutu here is an example of splitting signals into their own context so that each context (`TracesContext`, `LogsContext` in the example) don't need to share an interface and therefore no extraneous functions are added: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/15923.

Reply:
just change `IsContext` to `isContext` if that is what you want.